### ISSUE TITLE
Record label symbols are not found by Inline Rename if they follow a dot

### ISSUE THREAD

**Issue by cartermp** [Microsoft/visualfsharp#3493]
VS 2017 15.3.2.

Given the following code:

```fsharp
type Record = { TheInt: int; TheString: string }
type Foo = Bar of int | Baz of Record
let r1 = { TheInt = 12; TheString = "yes" }
let f1 = Baz { TheInt = 12; TheString = "bug" }

let f (r: Record) =
    if r.TheInt > 2 then () else ()
```

Invoke Inline Rename on any of the `TheInt` symbols:

![inline-rename-dot-record](https://user-images.githubusercontent.com/6309070/29642647-df893d6a-881e-11e7-9b36-06b6ac797ecc.PNG)

Note that the symbol in `r.TheInt` is not found.

**Comment by cartermp** [comment]
@abelbraaksma, in the case you mentioned earlier, I was actually able to get it to pick up the field inside of a DU pattern match with this code in a script file:

```
type Record = { TheInt: int; TheString: string }
type Foo = Bar of int | Baz of Record
let r1 = { TheInt = 12; TheString = "yes" }
let f1 = Baz { TheInt = 12; TheString = "bug" }

let g (f: Foo) =
    match f with
    | Baz ({TheInt = 3; TheString = "heya"} as x) -> x.TheString

let f (r: Record) =
    if r.TheInt > 2 then () else ()
```

![inline-rename-pm-works](https://user-images.githubusercontent.com/6309070/29642755-8fbb59d4-881f-11e7-8a2d-911f97e393a1.PNG)

Do you have a repro for this case that you could share?

**Comment by abelbraaksma** [comment]
@cartermp, at first it didn't work at all anymore when I tried to repro in a new project (i.e., nothing was renamed at all), but then I compiled and it seems it only works on compiled code to begin with. 

Then couldn't repro the part from issue #3462 anymore where I mentioned it happening in pattern matches. I may have misinterpreted my search result (it was in a large project) and there was code commented out that had the same name.

Which means that this is down to:

* Rename doesn't work at all unless you compile first
* Dotted notation on variables does not work

So, the good news:

* FQN notation works as expected
* Abbreviated notation works as expected
* Any other syntax (like left-hand pattern or assign expressions) I could think of (see screenshot) works

As an aside, on some occasions this dialog box popped up and waited rather long (30 - 60s). But again, this is a large project.

![image](https://user-images.githubusercontent.com/16015770/29650402-1acaee60-889b-11e7-968e-50f9d3bc8885.png)

The red-underlined in the screenshot below are the ones the rename services misses (and any comments, I didn't test strings):

![image](https://user-images.githubusercontent.com/16015770/29650456-5ac13baa-889b-11e7-8db3-55da8aea0c2e.png)
